File: 8 kyu/Total pressure calculation/solution.py
def solution(M1, M2, m1, m2, V, T) :
    return (m1/M1+m2/M2)*0.082*(T+273.15)/V
  
##############
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp) :
    
# 1 - Calculate the amount of substance in moles
    n1 = given_mass1 / molar_mass1
    n2 = given_mass2 / molar_mass2
# 2 - Convert the temperature from Celcius to kelvin  
    Temp_Kelvin = temp + 273.15
# 3 - Enter the gas constant
    R = 0.082
# 4 - Calculate the ideal gas law       
    P1 = (n1 * R * Temp_Kelvin) / volume
    P2 = (n2 * R * Temp_Kelvin) / volume
# 5 - Apply Dalton's law
    P = P1 + P2
    return P
  
###############
def solution(M1, M2, m1, m2, V, T):
    return (m1 / M1 + m2 / M2) * (T + 273.15) * .082 / V
  
###################
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp) :
    t = temp + 273.15
    R = 0.082
    n_mol1 = given_mass1 / molar_mass1
    n_mol2 = given_mass2 / molar_mass2
    return (n_mol1 * R * t/volume) + (n_mol2 * R * t/volume)
    
##################
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp) :
    p_total = ( ((given_mass1/molar_mass1)+(given_mass2/molar_mass2)) * (0.082) * (temp+273.15) )/ volume
    return p_total
  
###################
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp) :
    temp_k = temp + 273.15
    r = 0.082
    return (((given_mass1 / molar_mass1) + (given_mass2 / molar_mass2)) * r * temp_k) / volume 
  
#################
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp):
    return (given_mass1/molar_mass1 + given_mass2/molar_mass2)*.082*(temp + 273.15)*volume**(-1)
  
##############
def solution(molar_mass1, molar_mass2, given_mass1, given_mass2, volume, temp) :
    T=temp+273.15
    R=0.082
    P= (((given_mass1/molar_mass1)+(given_mass2/molar_mass2))*R*T)/volume
    return P

File: 8 kyu/Total pressure calculation/solution.c
double total_pressure(double mM1, double mM2, double gM1, double gM2, double v, double t) {
    return (gM1 / mM1 + gM2 / mM2) * 0.082 * (t + 273.15) / v;
}

#####################
double total_pressure(double M1, double M2, double m1, double m2, double V, double t) {

    double T=t+273.15;
    double R=0.082;
    return (m1/M1+m2/M2)*R*T/V;

}

####################
double total_pressure(double mM1, double mM2, double gM1, double gM2, double v, double t) 
{
    static double R = 0.082;
  
    double tK = 273.15 + t;
      
    return (gM1 / mM1 + gM2 / mM2) * R * tK / v;
}

###############
double total_pressure(double molar_mass1, double molar_mass2, double given_mass1, double given_mass2, double v, double t) {
double R = 0.082;
double T = t + 273.15;
double A = (given_mass1/molar_mass1);
double B = (given_mass2/molar_mass2);
double C = (A+B);
double D = (R*T);
double E = (C*D);
double F = E/v;

return F;//  <----  hajime!

}

################
double total_pressure(double M1, double M2, double m1, double m2, double v, double t) {
  return (m1 / M1 + m2 / M2) * 0.082 * (273.15 + t) / v;
}

#####################
double total_pressure(double molar_mass1, double molar_mass2, double given_mass1, double given_mass2, double v, double t) {
    double a = given_mass1 * 0.001 / molar_mass1;
    double b = given_mass2 * 0.001 / molar_mass2;
    t += 273.15;
    return (((a + b) * 0.082 * t) / v) * 1000;
}

###############
double total_pressure(double molar_mass1, double molar_mass2, double given_mass1, double given_mass2, double v, double t) {
  double res = 0;
  double R = 0.082;
  double T = t + 273.15;
  
  res = ((given_mass1 / molar_mass1) + (given_mass2 / molar_mass2)) * R * T / v;

  return res;
}
